textnow_bot.py

```diff
@@ -27,7 +29,7 @@ def __init__(self, page, cookies=None, username=None, password=None):
             raise Exception("missing authentication info")
 
         if "/messaging" not in page.url:
-            raise Exception("authentication failed")
+            raise Exception("authentication failed (did not reach textnow.com/messaging")
```

Comment:
Personally, I prefer to use colons to delimit exception descriptions from exception messages.

```suggestion
            raise Exception(
                "authentication failed: did not reach textnow.com/messaging"
            )
```